azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+| PATCH / PUT | if-match | value of etag     | value of if-match header DOES NOT equal the latest etag value on the server, indicating a change has ocurred since after the client fetched the resource|  412 Precondition Failed | Response body SHOULD return the serialized value of the resource (typically JSON) that was passed along with the request.|
+| DELETE      | if-none-match | value of etag     | value does NOT match the latest value on the server | 412 Preconditioned Failed | Response body SHOULD be empty.|
+| DELETE      | if-none-match | value of etag     | value matches the latest value on the server | 204 No Content | Response body SHOULD be empty.  |
+
+#### Computing ETags
+The strategy that you use to compute the ```ETag``` depends on its semantic. For example, it is natural, for resources that are inherently versioned, to use the version as the value of the ```ETag```. Another common strategy for determining the value of an ```ETag``` is to use a hash of the resource. If a resource is not versioned, and unless computing a hash is prohibitively expensive, this is the preferred mechanism. 
+
+> :ballot_box_with_check: **YOU SHOULD**, if using a hash strategy, hash the entire resource.
+
+> :heavy_check_mark: **YOU MAY** use or, include, a timestamp in your resource schema. If you do this, the timestamp shouldn't be returned with more than subsecond precision, and it SHOULD be consistent with the data and format returned, e.g. consistent on milliseconds.
+>
+> :heavy_check_mark: **YOU MAY** consider Weak ETags if you have a valid scenario for distinguishing between meaningful and cosmetic changes or if it is too expensive to compute a hash.
+
+### Distributed Tracing & Telemetry
+Azure SDK client guidelines specify that client libraries must send telemetry data through the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry. 
+Client libraries are required to send telemetry and distributed tracing information on every  request. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
+
+> :white_check_mark: **DO** follow the Azure SDK client guidelines for supporting telemetry headers and Open Telemetry.
+
+> :no_entry: **DO NOT** reject a call if you have custom headers you don't understand, and specifically, distributed tracing headers. 
+#### Additional References
+* [Azure SDK client guidelines](https://azure.github.io/azure-sdk/general_azurecore.html)
+* [Azure SDK User-Agent header policy](https://azure.github.io/azure-sdk/general_azurecore.html#azurecore-http-telemetry-x-ms-useragent)
+* [Azure SDK Distributed tracing policy](https://azure.github.io/azure-sdk/general_azurecore.html#distributed-tracing-policy) 
+* [Open Telemetry](https://opentelemetry.io/)
+
+## Final thoughts
```

Comment:
It's possible that I missed it, but I think we should have a section on granularity of service operations. From time to time we run into cases where a single user scenario corresponds to too many service calls. This means that client library has to to many calls to the service to actually do what the user cares about. This in turn is difficult to make reliable, explainable, etc. This issue could be mitigated by upleveling REST APIs or by providing batching APIs.

Reply:
I think there are two distinct thoughts in here. There first is the "upleveling" of an API. This is probably worth adding to the "Considerations" document. The second is batching, which feels like an new entry in the API patterns section. I'll add issues for both so we can review them with the larger team. 
See: #269 
See: #270